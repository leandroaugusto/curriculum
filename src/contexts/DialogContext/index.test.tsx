import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { DialogProvider, DialogContext } from ".";

describe("DialogProvider", () => {
  it("renders the DialogProvider", () => {
    const { container, getByTestId } = render(
      <DialogProvider>
        <DialogContext.Consumer>
          {({ toggle }) => <button data-testid="mock-share" onClick={toggle} />}
        </DialogContext.Consumer>
      </DialogProvider>
    );

    fireEvent.click(getByTestId("mock-share"));

    expect(container).toMatchSnapshot();
  });
});
