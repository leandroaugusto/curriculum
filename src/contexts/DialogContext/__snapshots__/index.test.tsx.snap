// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogProvider renders the DialogProvider 1`] = `
<div>
  <button
    data-testid="mock-share"
  />
</div>
`;
